Bitbucket server post build status 0.0.3

teameh opened the pull request:
![TagCheck](https://bitrise-steplib-git-check.herokuapp.com/tag?pr=2336)

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [X] __I will not move an already shared step version's tag to another commit__
- [X] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [ ] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [X] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [X] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

teameh commented:
Changes since https://github.com/bitrise-io/bitrise-steplib/pull/2319

- Changed script to except either a 0 or a 1 for bitrise status
- Added a new input for flagging to send bitbucket an INPROGRESS status 
- Added more logging

lpusok commented on steps/bitbucket-server-post-build-status/0.0.3/step.yml:
```diff
@@ -0,0 +1,73 @@
+- app_title: $BITRISE_APP_TITLE
+  opts:
+    is_required: true
+    title: Bitrise apptitle
```
typo in "title: Bitrise apptitle" should be Bitrise app title

lpusok commented on steps/bitbucket-server-post-build-status/0.0.3/step.yml:
```diff
@@ -0,0 +1,73 @@
+    is_sensitive: true
+    summary: Password for username
+    title: Bitbucket Server password
+- build_is_in_progress: false
```
In case a fixed list of inputs are accepted, use `value_options` to list them (e.g. https://github.com/bitrise-steplib/steps-gitlab-status/blob/94d495726810cad4963dde32267af6cead580f02/step.yml#L78); Use "yes" and "no" instead of true/false.

teameh replied:
Fixed

teameh commented on steps/bitbucket-server-post-build-status/0.0.3/step.yml:
```diff
@@ -0,0 +1,73 @@
+    is_required: true
+    summary: Full domain name without protocol eg 'my-domain.com'
+    title: Bitbucket Server domain name
+- bitbucket_server_username: null
```
Not all versions of bitbucket server feature personal access tokens. Support both options would be better but I don't have a bitbucket server instance supporting the tokens to test this atm. I could add this later if that is okay?

lpusok replied:
Sounds reasonable.

teameh commented on steps/bitbucket-server-post-build-status/0.0.1/step.yml:
```diff
@@ -0,0 +1,68 @@
+title: Bitbucket server post build status
```
Check. I'll delete the others

teameh commented on steps/bitbucket-server-post-build-status/0.0.3/step.yml:
```diff
@@ -0,0 +1,73 @@
+    entry_file: step.sh
+is_requires_admin_user: false
+is_always_run: true
+is_skippable: false
```
I thought users would want to know if this step is failing. Could I achieve this by omitting this parameter?

lpusok replied:
You are right, this status is only required for notification type steps, it is all right if set to false, this should be the same as not specified.

teameh replied:
Clear. Thanks

teameh commented on steps/bitbucket-server-post-build-status/0.0.3/step.yml:
```diff
@@ -0,0 +1,73 @@
+    is_sensitive: true
+    summary: Password for username
+    title: Bitbucket Server password
+- build_is_in_progress: false
```
Fixed

teameh commented:
Thanks for the review. I've addressed / responded to all your points. 

I've created a 0.0.6 tag and pushed the repo and updated this branch to include the 0.0.6 step

Now that the inputs have really simple name, might they not collide with other var's? I'm already using the step in my test setup and now I'm getting back a 401 from bitbucket server..

lpusok commented:
> Thanks for the review. I've addressed / responded to all your points.
> 
> I've created a 0.0.6 tag and pushed the repo and updated this branch to include the 0.0.6 step
> 
> Now that the inputs have really simple name, might they not collide with other var's? I'm already using the step in my test setup and now I'm getting back a 401 from bitbucket server..

Inputs should override already present env vars.
it could be debugged by adding `set -ex` or echoing env vars, with `set -ex` the only issue may be that the password may be logged.
I do not think there is a hard official input naming policy except for the format (there is a new Step Development Guideline in the works) so the `bitbucket` prefix is acceptable in case it does not work out.

lpusok commented on steps/bitbucket-server-post-build-status/0.0.6/step.yml:
```diff
@@ -0,0 +1,85 @@
+    - INPROGRESS
+    - SUCCESSFUL
+    - FAILED
+  preset_status: auto
```
`auto` is lowercase, usually it matters to be the same case as a valid value_option

teameh replied:
Shit, copy pasted that. Good catch, will fix

teameh commented:
- Fixed the env vars, my password was revoked, changing it fixed it. My bad. 
- Uppercased the "AUTO". I choose for uppercase because bitbucket expects the status in uppercase so I made all value options uppercase 
- Updated the readme to match the new inputs
- Created a 0.0.8 tag

teameh commented:
Hard to see the changes between original proposal and the things I changed after your review. Here's a comparison: https://gist.github.com/teameh/473ce9c049a2f85a06ceec04cde5b127/revisions

godrei commented on steps/bitbucket-server-post-build-status/0.0.8/step.yml:
```diff
@@ -0,0 +1,85 @@
+  password: null
+- opts:
+    description: |-
+      If set, this step will set a specific status instead of reporting the current build status.
```
`If set,` is not correct as input with value options, needs to have default value, please fix the description in the next step release.

godrei commented on steps/bitbucket-server-post-build-status/0.0.8/step.yml:
```diff
@@ -0,0 +1,85 @@
+  preset_status: AUTO
+- git_clone_commit_hash: $GIT_CLONE_COMMIT_HASH
+  opts:
+    is_required: true
```
might `is_dont_change_value: true` would fit for this input too.
